algorithms/src/polycommit/kzg10/mod.rs

```diff
@@ -96,121 +93,11 @@ pub struct KZG10<E: PairingEngine>(PhantomData<E>);
-        end_timer!(gamma_g_time);
-
-        // Reduce `beta^i G` and `gamma beta^i G` to affine representations.
-        let powers_of_beta_g = E::G1Projective::batch_normalization_into_affine(powers_of_beta_g);
-        let powers_of_beta_times_gamma_g =
-            E::G1Projective::batch_normalization_into_affine(powers_of_beta_times_gamma_g)
-                .into_iter()
-                .enumerate()
-                .collect();
-
-        // This part is used to derive the universal verification parameters.
-        let list = supported_degree_bounds_config.get_list::<E::Fr>(max_degree);
-
-        let supported_degree_bounds =
-            if *supported_degree_bounds_config != KZGDegreeBounds::None { list.clone() } else { vec![] };
-
-        // Compute `neg_powers_of_beta_h`.
-        let inverse_neg_powers_of_beta_h_time = start_timer!(|| "Generating negative powers of h in G2");
-        let inverse_neg_powers_of_beta_h =
-            if produce_g2_powers && *supported_degree_bounds_config != KZGDegreeBounds::None {
-                let mut map = BTreeMap::<usize, E::G2Affine>::new();
-
-                let mut neg_powers_of_beta = vec![];
-                for i in list.iter() {
-                    neg_powers_of_beta.push(beta.pow([(max_degree - *i) as u64]).inverse().unwrap());
-                }
-
-                let window_size = FixedBase::get_mul_window_size(neg_powers_of_beta.len());
-                let neg_h_table = FixedBase::get_window_table(scalar_bits, window_size, h);
-                let neg_powers_of_h =
-                    FixedBase::msm::<E::G2Projective>(scalar_bits, window_size, &neg_h_table, &neg_powers_of_beta);
-
-                let affines = E::G2Projective::batch_normalization_into_affine(neg_powers_of_h);
-
-                for (i, affine) in list.iter().zip_eq(affines.iter()) {
-                    map.insert(*i, *affine);
-                }
-
-                map
-            } else {
-                BTreeMap::new()
-            };
-        end_timer!(inverse_neg_powers_of_beta_h_time);
-
-        let beta_h = h.mul(beta).to_affine();
-        let h = h.to_affine();
-        let prepared_h = h.prepare();
-        let prepared_beta_h = beta_h.prepare();
-
-        let powers = PowersOfG::<E>::setup(powers_of_beta_g, powers_of_beta_times_gamma_g)?;
-        let pp = UniversalParams {
-            powers: Arc::new(RwLock::new(powers)),
-            h,
-            beta_h,
-            supported_degree_bounds,
-            inverse_neg_powers_of_beta_h,
-            prepared_h,
-            prepared_beta_h,
-        };
-        end_timer!(setup_time);
-        Ok(pp)
+    pub fn load_srs(max_degree: usize) -> Result<UniversalParams<E>, PCError> {
+        let params = UniversalParams::load()?;
+        params.download_powers_for(0..(max_degree + 1))?;
+        // TODO: check that params match supported_degree bounds.
```

Comment:
Is this TODO needed?

Reply:
No, removed in [`1e21f75` (#1245)](https://github.com/AleoHQ/snarkVM/pull/1245/commits/1e21f7525c2d54d68146fada724bc611092b75f7)